### PR TITLE
Rename `dont_need` to `discard` and mention `decommit` as an option

### PR DESCRIPTION
**Pull request by lukewagner**
As discussed in #384.

### CODE REVIEW

File: FutureFeatures.md [hunk 1]
```diff
@@ -22,17 +22,19 @@ Provide access to safe OS-provided functionality including:
...
+   POSIX)
 * `shmem_create(length)`: create a memory object that can be simultaneously
   shared between multiple linear memories
 * `map_shmem(addr, length, shmem, shmem-offset)`: like `map_file` except
   `MAP_SHARED`, which isn't otherwise valid on read-only Blobs
 * `mprotect(addr, length, prot-flags)`: change protection on the range
   `[addr, addr+length)` (where `addr+length <= memory_size`)
+* `decommit(addr, length)`: equivalent to `mprotect(addr, length, PROT_NONE)`
+  followed by `discard(addr, length)` and potentially more efficient than
+  performing these operations in sequence.
```
Comment (jfbastien) [inline FutureFeatures.md]: Getting `PROT_NONE` has a bigger meaning that decommitting IMO. I'm not sure what name would be better... "void" has other meanings.

Comment (lukewagner) [inline FutureFeatures.md]: I'm not sure what you mean.  Are you commenting on the naming of `decommit`?

Comment (jfbastien) [inline FutureFeatures.md]: Yeah, but without a better name to suggest I'll lgtm and hope someone else comes up with a nicer name before you commit :-)

Comment (lukewagner) [inline FutureFeatures.md]: Hah, thanks.  As we see here with `dont_need`=>`discard`, we can always rename.

**Pull request merged**